src/main/java/com/sublinks/sublinksapi/api/lemmy/v3/comment/services/LemmyCommentReplyService.java

```diff
@@ -46,40 +56,50 @@ private CommentReplyView.CommentReplyViewBuilder commentReplyViewBuilder(
     final Comment lemmyComment = conversionService.convert(commentReply.getComment(),
         Comment.class);
 
-    final CommentAggregates commentAggregates = conversionService.convert(
-        commentReply.getComment().getCommentAggregate(), CommentAggregates.class);
+    final CommentAggregates commentAggregates = conversionService.convert(commentReply.getComment()
+        .getCommentAggregate(), CommentAggregates.class);
 
-    final Post lemmyPost = conversionService.convert(commentReply.getComment().getPost(),
-        Post.class);
+    final Post lemmyPost = conversionService.convert(commentReply.getComment()
+        .getPost(), Post.class);
 
-    final Community lemmyCommunity = conversionService.convert(
-        commentReply.getComment().getCommunity(), Community.class);
+    final Community lemmyCommunity = conversionService.convert(commentReply.getComment()
+        .getCommunity(), Community.class);
+    final boolean creatorBannedFromCommunity = linkPersonCommunityService.hasLink(creator,
+        commentReply.getComment()
+            .getCommunity(), LinkPersonCommunityType.banned);
 
-    final int personVote = commentLikeService.getPersonCommentVote(Person,
+    final int personVote = commentLikeService.getPersonCommentVote(person,
         commentReply.getComment());
 
+    boolean creatorIsModerator = false;
+    if (!creatorBannedFromCommunity) {
```

Comment:
Can you ban an admin? If not, then you could move the check for isAdmin here too and save some DB calls for non-admin users

Do we allow blocking an admin or mod? If not, then you should skip that check. Also, if a user is banned, does it matter if they are blocked? (Or vice-versa, those checks could be done in either order but they render the other one unneeded if either is "true")

Reply:
We dont allow blocking an admin.